### PR TITLE
Target: refactoring function verify_fix_path

### PR DESCRIPTION
**Pull request by otaviobp**

### CODE REVIEW

File: src/target_root.c [hunk 1]
```diff
@@ -209,7 +209,10 @@ static enum swupd_code stage_single_file(struct file *file, struct manifest *mom
 		debug("Target directory does not exist: %s\n", target_path);
 		if (mom) {
 			debug("Attempting to fix missing directory\n");
-			verify_fix_path(dir, mom);
+			if (!verify_fix_path(dir, mom)) {
```
Comment (castulo) [inline src/target_root.c:212]: if you are checking the success of creating the directory here then we don't need to check again in line 225

File: src/target_root.c [hunk 2]
```diff
@@ -309,119 +312,117 @@ static enum swupd_code stage_single_file(struct file *file, struct manifest *mom
...
+			goto end;
+		}
+		debug("Corrupt directory: %s\n", target_path);
+	} else {
+		debug("Missing directory: %s\n", target_path);
+	}
+
+	/* In some circumstances (Docker using layers between updates/bundle adds,
+	 * corrupt staging content) we could have content which fails to stage.
+	 * In order to avoid this causing failure in verify_fix_path, remove the
```
Comment (castulo) [inline src/target_root.c:350]: typo, "this cause of failure"?

File: src/target_root.c [hunk 3]
```diff
@@ -309,119 +312,117 @@ static enum swupd_code stage_single_file(struct file *file, struct manifest *mom
...
-		if (ret != 0) {
-			/* stage_single_file returns a swupd_code on error,
-			* just propagate the error */
-			debug("Path %s failed to stage in verify_fix_path\n", path);
-			goto end;
+	/* Loop parents */
+	for (i = list_dir; i; i = i->next) {
+		cur_dir = i->data;
+
+		// Abort of first error
```
Comment (castulo) [inline src/target_root.c:417]: typo "of" -> "on"

File: src/bundle_add.c [hunk 4]
```diff
@@ -240,7 +240,6 @@ static enum swupd_code install_files(struct manifest *mom, struct list *to_insta
 	progress_next_step("install_files", PROGRESS_BAR);
 
 	//TODO: Improve staging functions so we won't need this hack
```
Comment (castulo) [inline src/bundle_add.c:242]: you can remove this comment now too

**Comment by otaviobp** [comment]
V2: Fixing reported problems

File: src/target_root.c (hunk 1 above)
Comment (castulo) [reply src/target_root.c:292]: you missed this comment

Comment (castulo) [reply src/target_root.c:292]: ahh maybe not, looks like my review hadn't refreshed yet

**Pull request merged**